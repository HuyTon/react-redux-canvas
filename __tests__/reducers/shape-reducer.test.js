// Reducer to be tested
import shapeReducer from '../../src/client/reducers/shape-reducer';
import {
  GET_SHAPES_SUCCESS,
  DRAW_FULL_SHAPE,
  DRAW_SEMI_SHAPE,
  REMOVE_SHAPE
} from '../../src/client/actions/action-types';

describe('GET_SHAPES_SUCCESS', () => {
  test('is correct', () => {
    const action = { type: GET_SHAPES_SUCCESS };

    expect(shapeReducer(undefined, action)).toMatchSnapshot();
  });
});

describe('DRAW_FULL_SHAPE', () => {
  test('is correct', () => {
    const action = { type: DRAW_FULL_SHAPE };

    expect(shapeReducer(undefined, action)).toMatchSnapshot();
  });
});

describe('DRAW_SEMI_SHAPE', () => {
  test('is correct', () => {
    const action = { type: DRAW_SEMI_SHAPE };

    expect(shapeReducer(undefined, action)).toMatchSnapshot();
  });
});

describe('REMOVE_SHAPE', () => {
  test('is correct', () => {
    const action = { type: REMOVE_SHAPE };

    expect(shapeReducer(undefined, action)).toMatchSnapshot();
  });
});