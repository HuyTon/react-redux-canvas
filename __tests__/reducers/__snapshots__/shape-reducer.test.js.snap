// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DRAW_FULL_SHAPE is correct 1`] = `
Object {
  "shapes": Object {
    "shapes": Array [
      Object {
        "active": true,
        "fullShape": true,
      },
    ],
  },
}
`;

exports[`DRAW_SEMI_SHAPE is correct 1`] = `
Object {
  "shapes": Object {
    "shapes": Array [
      Object {
        "active": true,
        "fullShape": false,
      },
    ],
  },
}
`;

exports[`GET_SHAPES_SUCCESS is correct 1`] = `
Object {
  "shapes": undefined,
}
`;

exports[`REMOVE_SHAPE is correct 1`] = `
Object {
  "shapes": Object {
    "shapes": Array [
      Object {
        "active": false,
        "fullShape": false,
      },
    ],
  },
}
`;
